sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_session_receiver.py

```diff
@@ -0,0 +1,167 @@
+    :keyword str queue_name: The path of specific Service Bus Queue the client connects to.
+    :keyword str topic_name: The path of specific Service Bus Topic which contains the Subscription
+     the client connects to.
+    :keyword str subscription_name: The path of specific Service Bus Subscription under the
+     specified Topic the client connects to.
+    :keyword int prefetch: The maximum number of messages to cache with each request to the service.
+     The default value is 0, meaning messages will be received from the service and processed
+     one at a time. Increasing this value will improve message throughput performance but increase
+     the change that messages will expire while they are cached if they're not processed fast enough.
+    :keyword float idle_timeout: The timeout in seconds between received messages after which the receiver will
+     automatically shutdown. The default value is 0, meaning no timeout.
+    :keyword mode: The mode with which messages will be retrieved from the entity. The two options
+     are PeekLock and ReceiveAndDelete. Messages received with PeekLock must be settled within a given
+     lock period before they will be removed from the queue. Messages received with ReceiveAndDelete
+     will be immediately removed from the queue, and cannot be subsequently rejected or re-received if
+     the client fails to process the message. The default mode is PeekLock.
+    :paramtype mode: ~azure.servicebus.ReceiveSettleMode
+    :keyword session_id: A specific session from which to receive. This must be specified for a
+     sessionful entity, otherwise it must be None. In order to receive messages from the next available
+     session, set this to None.  The default is None.
+    :paramtype session_id: str
+    :keyword bool logging_enable: Whether to output network trace logs to the logger. Default is `False`.
+    :keyword int retry_total: The total number of attempts to redo a failed operation when an error occurs.
+     Default value is 3.
+    :keyword transport_type: The type of transport protocol that will be used for communicating with
+     the Service Bus service. Default is `TransportType.Amqp`.
+    :paramtype transport_type: ~azure.servicebus.TransportType
+    :keyword dict http_proxy: HTTP proxy settings. This must be a dictionary with the following
+     keys: `'proxy_hostname'` (str value) and `'proxy_port'` (int value).
+     Additionally the following keys may also be present: `'username', 'password'`.
+
+    .. admonition:: Example:
+
+        .. literalinclude:: ../samples/sync_samples/sample_code_servicebus.py
+            :start-after: [START create_servicebus_receiver_sync]
+            :end-before: [END create_servicebus_receiver_sync]
+            :language: python
+            :dedent: 4
+            :caption: Create a new instance of the ServiceBusReceiver.
+
+    """
+    def __init__(self, fully_qualified_namespace, credential, **kwargs):
```

Comment:
Open question.  Can theoretically unify this with async version in the session mixin, but given that it provides the external APIs, and we offer bolstered docstrings for the async compat version, I put it off. (and that it would basically turn the session receiver into an empty composite type)

Reply:
from the view point of docstrings/code snippets, I'm good with not unifying them as there is barely worth being reused.